public/app/screens/main/sections/devices/device/device.jsx

```diff
@@ -1,19 +1,22 @@
 var React = require('react');
+var Link = require('react-router').Link;
 var styleMixin = require('mixins/style-mixin');
 var SwitchControl = require('./controls/switch');
 var SliderControl = require('./controls/slider');
+var DeviceState = require('./../device-state');
 
 var Device = React.createClass({
-  mixins: [ styleMixin(require('./style.scss')) ],
+  mixins: [styleMixin(require('./style.scss'))],
 
   render: function render () {
     return (
       <div className="cc-device">
-        <div className="cc-device__title">{this.props.device.title}</div>
+        <Link className="cc-device__title" to="device" params={{deviceId: this.props.device.id}}>
+          {this.props.device.title}
+        </Link>
         <div className="cc-device__type">{this.props.device.type}</div>
         <div className="cc-device__control">{this.getControl()}</div>
-        <div className={'cc-device__state-image ' + this.getStateImageClass()}/>
-        <div className="cc-device__state-text">{this.getConnectedText()}</div>
+        <DeviceState state={this.props.device.state}/>
```

Comment:
Better pass the `device` itself to the component. It better allows changes if we once decide to get `state` information from other `device`'s properties as well.